server/api/lib/tags/utils.js

```diff
@@ -0,0 +1,33 @@
+const { tagDataRules } = require('../../constants');
+const PlanAreaChanges = require('../../../api/model/plan_area_changes');
+
+const isTagByUsageAddition = async (planId, tag, rule) => {
+	try {
+		const result = await PlanAreaChanges.byPlanAndUsage(planId, tagDataRules[rule].usage);
+		if (result && result.models && result.models[0]) {
```

Comment:
Also, I think it'd be better if the rule description should be done in the housing file, and the isTagByUsageAddition function should also return true/ false.

For example, send to the function `isTagByUsageAddition` what it needs to check, and it'll return true/ false. Then the housing will add it .

In this case, the `isTagByUsageAddition` function does not know the logic of the rules, and only performs what it's meant to do - check a specific land use change. Also, it will not require sending it the tag info

```
if (housingByUnits) {
		dataRules.push(
			created_by_data_rules: {   
						tag_id: tag, 
						created_by_data_rules: `{rule:'${tagDataRules[rule].description}',detail:'adds ${changeToApprovedState} ${tagDataRules[rule].usage}'}`
					}
		);
	}

```